api/src/main/proto/stellarstation/api/v1/stellarstation.proto

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2018 Infostellar, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package stellarstation.api.v1;
+
+option java_multiple_files = true;
+option java_outer_classname = "StellarstationProto";
+option java_package = "com.stellarstation.api.v1";
+
+// The public API service of Stellarstation, a system supporting communication between satellites
+// and antennas. This API is for use by an operator of a satellite - it is illegal to specify
+// satellites that the operator does not own or passes they have not reserved.
+// Only using IDs shown on the StellarStation Console or returned in API responses will ensure all
+// inputs are valid.
+service StellarStationService {
+  // Open a stream to a satellite. The returned stream is bi-directional - it can be used by the
+  // client to send commands to the satellite and data received from the satellite will be returned
+  // as it is made available. All telemetry received from the satellite from this point on will be
+  // returned as soon as this method is called. If `SatelliteStreamRequest.stream_id` is specified,
+  // any messages that have been buffered for the stream will be returned as well.
+  //
+  // The stream can be left open for any amount of time and telemetry will be received as available
+  // and returned. When the client is done with the stream, it should close it successfully. If the
+  // stream is cancelled with an unexpected error at any point, telemetry received before the next
+  // call to `OpenSatelliteStream` to resume the stream will be buffered up to 1 minute, and if no
+  // call to `OpenSatelliteStream` is made with `stream_id` specified to resume the stream, it will
+  // expire in 10 minutes and a call to `OpenSatelliteStream` with that `stream_id` will be closed
+  // with a `ABORTED` error.
+  //
+  // If the satellite is not found or the client is not authorized for it, the stream will be closed
+  // with a `NOT_FOUND` error.
+  rpc OpenSatelliteStream (stream SatelliteStreamRequest) returns (stream SatelliteStreamResponse);
```

Comment:
Dumb question :) What happens if I open a stream to a satellite but there is no active plan at the moment? I assume this whole API is based around the idea that the client only calls it when a plan has started?

Reply:
Nah the API is just to get any data for a satellite as it arrives. It's also why I mentioned the stream can be left open for any amount of time - it would be idle until a plan started but it's ok. Tried adding some words to see if it's clearer that way.